### ISSUE TITLE
Crash with Hessians (Bus Error: 10)

### ISSUE THREAD

**Issue by arnavs** [SciML/ModelingToolkit.jl#350]
Here's an example: 

```
julia> H[1]
c ^ -1 - (1 + -1δ + exp(z_p) * α * k_p ^ (-1 + α)) * c_p ^ -1 * β

julia> X_p
8-element Array{Operation,1}:
 c_p
 y_p
   c
   y
 k_p
 z_p
   k
   z

julia> ModelingToolkit.hessian(H[1], X_p)
Bus error: 10
```

**Comment by shashi** [comment]
```julia
julia> ModelingToolkit.hessian(h, z)
8×8 Array{Expression,2}:
 -2 * (1 + -1δ + exp(z_p) * α * k_p ^ (-1 + α)) * abs2(inv(c_p)) * inv(c_p) * β  …  Constant(0)  Constant(0)
                                                                    Constant(0)     Constant(0)  Constant(0)
                                                                    Constant(0)     Constant(0)  Constant(0)
                                                                    Constant(0)     Constant(0)  Constant(0)
                  abs2(inv(c_p)) * exp(z_p) * (-1 + α) * β * α * k_p ^ (-2 + α)     Constant(0)  Constant(0)
                             abs2(inv(c_p)) * exp(z_p) * β * α * k_p ^ (-1 + α)  …  Constant(0)  Constant(0)
                                                                    Constant(0)     Constant(0)  Constant(0)
                                                                    Constant(0)     Constant(0)  Constant(0)

```

It works for me unfortunately.....

**Comment by shashi** [comment]
Is there a way you can share the full code?

**Comment by arnavs** [comment]
OK, fixed for me on 3.2.0 symbolicutils and MTK: 

```
julia> ModelingToolkit.hessian(x, X_p)

8×8 Array{Expression,2}:
 -2 * (1 + -1δ + exp(z_p) * α * k_p ^ (-1 + α)) * abs2(inv(c_p)) * inv(c_p) * β  Constant(0)                Constant(0)  …  Constant(0)  Constant(0)
                                                                    Constant(0)  Constant(0)                Constant(0)     Constant(0)  Constant(0)
                                                                    Constant(0)  Constant(0)  2 * abs2(inv(c)) * inv(c)     Constant(0)  Constant(0)
                                                                    Constant(0)  Constant(0)                Constant(0)     Constant(0)  Constant(0)
                  abs2(inv(c_p)) * exp(z_p) * (-1 + α) * β * α * k_p ^ (-2 + α)  Constant(0)                Constant(0)     Constant(0)  Constant(0)
                             abs2(inv(c_p)) * exp(z_p) * β * α * k_p ^ (-1 + α)  Constant(0)                Constant(0)  …  Constant(0)  Constant(0)
                                                                    Constant(0)  Constant(0)                Constant(0)     Constant(0)  Constant(0)
                                                                    Constant(0)  Constant(0)                Constant(0)     Constant(0)  Constant(0)
```
